scripts/prepare_sector_network.py

```diff
@@ -4538,6 +4590,17 @@ def add_enhanced_geothermal(n, egs_potentials, egs_overlap, costs):
     )
     pop_weighted_energy_totals.update(pop_weighted_heat_totals)
 
+    if options.get("keep_existing_capacities", False):
```

Comment:
```suggestion
    if options.get("keep_existing_capacities", False) and snakemake.params.foresight not in ["myopic", "perfect"]:
```
It would make sense to limit this functionality to overnight, as for myopic and perfect optimization existing caps are added later on